### DISCUSSION TITLE
[FEATURE]: Add Empty Trash button

### DISCUSSION THREAD

**Discussion by arsinclair** [doublecmd/doublecmd#1727] [Suggestions]
Krusader has a great feature — Empty Trash command. Once clicked it empties the system trash.

![image](https://github.com/doublecmd/doublecmd/assets/2878904/bd05b238-a747-4de6-b4fd-39e1900a5584)

It would be great to see such a feature in DC. It might not be very useful for Windows users because they can already right-click on the Trash icon on the desktop and clear it. However, on my Debian it would be invaluable: there's no Trash Bin on the desktop, trash is just a folder located at `~/.local/share/Trash` and in order to empty it I have to go to the folder and delete stuff manually. This function in Krusader is the only reason I keep it installed on my system and cannot fully transition to DC.

Cheers!

**Comment by JonnyTech** [comment]
I think that you can right-click the F8 button at bottom right and a popup menu appears which has that option.

**Comment by alexx2000** [comment]
You can add such function yourself. Copy this button:
```
DOUBLECMD#TOOLBAR#XMLDATA<?xml version="1.0" encoding="UTF-8"?>
<doublecmd>
  <Program>
    <ID>{00FC0338-4293-4689-8335-6B6AEE07C289}</ID>
    <Icon>user-trash</Icon>
    <Hint>Empty trash</Hint>
    <Command>gio</Command>
    <Params> trash --empty</Params>
  </Program>
</doublecmd>
```
and paste it into the toolbar (via context menu). Requires `libglib2.0-bin` package.

@JonnyTech only under Windows.

**Reply by arsinclair** [reply]
@alexx2000 worked like a charm, thanks. Do you feel it can be added to DC OOTB?

**Comment by ags1234** [comment]
It would be nice to have a directory Scripts distributed with DC, where one could find a collection of interesting and useful stuff.
